### ISSUE TITLE
Requests act differently when created through tests vs from HTTP request

### ISSUE THREAD

**Issue by jameshulse** [laravel/framework#46455]
- Laravel Version: 9.50.2
- PHP Version: 8.2.2
- Database Driver & Version: Postgres 14.2

### Description:

TLDR: `$request->getContent()` is empty when using `$this->post(...)` in unit tests.

In our application code we are reading the raw content of a request with `$request->getContent()` and then later accessing the contents as JSON with e.g `$request->get('key')`. This code fails when running through our unit tests.

It seems that the test request created via e.g `$this->post(...)` doesn't have its raw content set based on the data passed. This differs to how requests are created when actually handled by the incoming HTTP stack as the request will have both content, and it's request InputBag set.

### Steps To Reproduce:

Controller code:
```php
public function exampleHandler(Request $request)
{
    $content = $request->getContent(); // Empty when called via post in unit tests
    $input = $request->get('foo');

    // TODO: Make use of content and input
}
```

Test code:
```php
public function test_exampleHandler()
{
    $this->post('example', ['foo' => 'bar']); // When calling exampleHandler "$input" will have a value, but $content won't
    
    $this->call('POST', 'example', content: '{ "foo": "bar" }'); // I would expect this to act the same, but $content will now have a value, but $input won't
}
```

If we posted the same raw body via Postman to our application then the `exampleHandler` would pickup both the raw content, and the input via `->get()`.

**Comment by ankurk91** [comment]
Who wants to use Apache/Ngnix in their test suites?

**Comment by jameshulse** [comment]
> Who wants to use Apache/Ngnix in their test suites?

No one I assume. I am talking about an HTTP request to my application under normal running circumstances as compared to a test against my controller using standard Laravel methods. If that wasn't clear the two cases are:

1. While my application is running, if I receive a request then I can access the raw content of the request with `$request->getContent()` and I can also get individual JSON values with `$request->get('...')`
2. When testing that same controller in a unit test, `$request->getContent()` will be empty despite passing a value into `$this->post`. I would expect the content to be a JSON encoded form of the data I passed.

**Comment by timacdonald** [comment]
It looks like you are not specifying that your content is JSON encoded. 

In POSTMAN are you sending an `Content-type: application/json`? If you are, you also need to send a JSON request in your unit tests...

```diff
- $response = $this->post(...);
+ $response = $this->postJson(...);
```

**Comment by jameshulse** [comment]
Hi @timacdonald. Thanks for your reply. It seems you are 100% correct. Changing my test to use `postJson` instead of `post` means that `$request->getContent()` now works correctly.

I do feel a bit silly for not trying this, but I have to admit that it wasn't very obvious at all that this is the case. I can now see how the `postJson` code handles setting the `content` on the request, and I will just have to remember this in the future.

Thanks again for taking the time to reply.

**Comment by timacdonald** [comment]
No troubles.

**Comment by mirko77** [comment]
Well, just in case anyone else ends up here, here is my findings.

If you're specifically trying to read the raw input sent as the request body using getContent(), ensure that the data is sent as raw content and not as a URL-encoded form. Here's an example:

`$response = $this->actingAs($user)->call('POST', 'api/endpoint/' . $project->slug, [], [], [], [], $rawContent);`
